### ISSUE TITLE
Cells created by VoronoiDiagramBuilder don't adhere to the given tolerance

### ISSUE THREAD

**Issue by jr-accurate** [locationtech/jts#350]
I'm having issues with the Polygons created by VoronoiDiagramBuilder. Specifically I'm trying to do some further calculations with them, which then fail with a TopologyException because the points in one of the Polygons are too close together: (30.669999999999998, 46.1645, NaN) and (30.67, 46.1645, NaN).

I did set the tolerance to different values, but that didn't change anything. I think the almost equal coordinates come from `QuadEdge.rot()`, but that may not be the root cause.

If there's a way to prevent this using the tolerance differently and/or a different PrecisionModel, I'm happy to learn ;)

Here's the smallest testcase I could come up with:
```
public class VoronoiErrorTest {

    @Test
    public void testVoronoi() {
        GeometryFactory geometryFactory = new GeometryFactory(new PrecisionModel(1_000_000));

        Polygon jtsPolygon = geometryFactory.createPolygon(new Coordinate[]{
            new Coordinate (-12.1, -1.9),
            new Coordinate (74.9, -1.9),
            new Coordinate (74.9, 53.9),
            new Coordinate (-12.1, 53.9),
            new Coordinate (-12.1, -1.9)
        });

        Envelope envelope = jtsPolygon.getEnvelopeInternal();
        VoronoiDiagramBuilder voronoiDiagramBuilder = new VoronoiDiagramBuilder();
        voronoiDiagramBuilder.setClipEnvelope(envelope);
        voronoiDiagramBuilder.setSites(getCoordinates());
        voronoiDiagramBuilder.setTolerance(0.00001);

        Geometry diagram = voronoiDiagramBuilder.getDiagram(geometryFactory);
        diagram.getGeometryN(2).intersection(jtsPolygon);
    }

    private List<Coordinate> getCoordinates(){
        return Arrays.asList(
            new Coordinate (31.0, 46.94),
            new Coordinate (30.9, 46.44),
            new Coordinate (30.44, 46.44),
            new Coordinate (30.39, 45.94),
            new Coordinate (30.46, 45.18),
            new Coordinate (30.95, 45.94)
        );
    }
}
```
Thanks!

**Comment by dr-jts** [comment]
This is a bug.  Not sure what the right fix should be.  Possibly use the tolerance value to snap the vertices of the computed cell polygons.

In the meantime, perhaps you can just round the vertices off using `GeometryPrecisionReducer`.

**Comment by dr-jts** [comment]
For ease of reproducing in the TestBuilder, here's the input data:

Sites: MULTIPOINT ((31 46.94), (30.9 46.44), (30.44 46.44), (30.39 45.94), (30.46 45.18), (30.95 45.94))
Clip Envelope: POLYGON ((-12.1 -1.9, 74.9 -1.9, 74.9 53.9, -12.1 53.9, -12.1 -1.9))

**Comment by dr-jts** [comment]
is this fixed by #448?

**Comment by jr-accurate** [comment]
As the example above works with 448, I'd say yes. I didn't have time to do more testing in our application yet, but so far I consider this fixed ;)

**Comment by dr-jts** [comment]
Fixed by #448
